src/databricks/labs/lsql/dashboards.py

```diff
@@ -728,6 +766,118 @@ def _get_query_widget_spec(fields: list[Field], *, frame: WidgetFrameSpec | None
         return spec
 
 
+@dataclass
+class FilterTile(Tile):
+    _position: Position = dataclasses.field(default_factory=lambda: Position(0, 0, 3, 2))
+
+    def validate(self) -> None:
+        """Validate the tile
+
+        Raises:
+            ValueError : If the tile is invalid.
+        """
+        if not self.metadata.is_filter():
+            raise ValueError(f"Tile is not a filter file: {self}")
+        if len(self.metadata.filters) == 0:
+            raise ValueError(f"Filter tile with path {self.metadata.path} has no filters defined")
+
+    def get_layouts(self, datasets: list[Dataset]) -> Iterable[Layout]:
+        """Get the layout(s) reflecting this tile in the dashboard."""
+        dataset_columns = self._get_dataset_columns(datasets)
+        if len(dataset_columns) == 0:
```

Comment:
Should be part of `validate`

Reply:
dashboard_metadata.validate() is called before _get_layouts is called. The dataset_columns is created during get layout. So it can't be validated in validate in current case. As this is a computed value why it needs to be validated as a metadata property?

Reply:
I see, the position makes sense then. Note that we do not warn when one of the columns in the filter columns list does not match but others do

Reply:
leave it as code comments to pre-emptively avoid review comments :)